llvm/lib/Target/AArch64/AArch64PointerAuth.cpp

```diff
@@ -359,6 +366,32 @@ bool AArch64PointerAuth::checkAuthenticatedLR(
   return true;
 }
 
+void AArch64PointerAuth::emitBlend(MachineBasicBlock::iterator MBBI,
+                                   Register Result, Register AddrDisc,
+                                   unsigned IntDisc) const {
+  MachineBasicBlock &MBB = *MBBI->getParent();
+  DebugLoc DL = MBBI->getDebugLoc();
+
+  if (Result != AddrDisc)
+    BuildMI(MBB, MBBI, DL, TII->get(AArch64::ORRXrs), Result)
```

Comment:
Looking at the one obvious regression test for the blend intrinsic, in "ptrauth-intrinsic-blend.ll", I seems to me that the case `Result != AddrDisc` isn't covered there, so the then part here isn't covered by a test case.
If my impression is correct, maybe it would make sense to add a test to that file to cover this case?